rotkehlchen/tasks/calendar.py

```diff
@@ -47,3 +70,101 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
+                group_by_event_ids=False,
+                filter_query=EvmEventFilterQuery.make(
+                    and_op=True,
+                    counterparties=[CPT_ENS],
+                    event_types=[event_type],
+                    event_subtypes=[event_subtype],
+                ),
+            ))
+
+    if len(ens_events) == 0:
+        return
+
+    db_calendar = DBCalendar(database=database)
+    customizable_date = CustomizableDateMixin(database=database)
+    with database.conn.read_ctx() as cursor:
+        blockchain_accounts = database.get_blockchain_accounts(cursor=cursor)
+
+    current_ts = ts_now()
+    ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+    for ens_event in ens_events:
+        if not (
+            ens_event.location_label is not None and
+            ens_event.extra_data is not None and
+            (ens_name := ens_event.extra_data.get('fullname')) is not None and
+            (ens_expires := ens_event.extra_data.get('expires')) is not None and (
+                user_address := string_to_evm_address(ens_event.location_label)
+            ) in blockchain_accounts.get(
+                blockchain := ChainID.deserialize(ens_event.location.to_chain_id()).to_blockchain(),  # noqa: E501
+            ) and (ens_expires := Timestamp(ens_expires)) > current_ts  # those with expiry in the future  # noqa: E501
+        ):
+            continue
+
+        if (calendar_entries := db_calendar.query_calendar_entry(
+            filter_query=CalendarFilterQuery.make(
+                and_op=True,
+                name=f'{ens_name} expiry',
+                addresses=[OptionalBlockchainAddress(
+                    blockchain=blockchain,
+                    address=user_address,
+                )],
+                blockchain=blockchain,
+                counterparty=CPT_ENS,
+            ),
+        ))['entries_found'] == 0:  # if calendar entry doesn't exist, add it
+            calendar_entry_id = db_calendar.create_calendar_entry(CalendarEntry(
+                name=f'{ens_name} expiry',
+                timestamp=ens_expires,
+                description=f'{ens_name} expires on {customizable_date.timestamp_to_date(ens_expires)}',  # noqa: E501
+                counterparty=CPT_ENS,
+                address=user_address,
+                blockchain=blockchain,
+                color=None,
+                auto_delete=True,
+            ))
+        else:  # else
+            calendar_entry_id = calendar_entries['entries'][0].identifier
+
+        if (
+            ens_name not in ens_to_event or
+            ens_expires > ens_to_event[ens_name][1].extra_data['expires']  # type: ignore[index]  # extra_data is not None, checked above
+        ):  # insert mapping for the latest expiry timestamp
+            ens_to_event[ens_name] = (calendar_entry_id, ens_event)
+
+    _, failed_to_add = db_calendar.create_reminder_enties(reminders=[
```

Comment:
1. What happens if reminder already exists?
2. The function you call here has a typo

`create_reminder_enties` -> `create_reminder_entries`

Reply:
if the reminder entry already exists then we don't add it for that calendar entry, because it's possible that the user could have already set their own custom reminder
it's checked by this line
```py
if db_calendar.count_reminder_entries(event_id=calendar_identifier) == 0  # if not already present  # noqa: E501
```